src/com/mateanna/hw5/TestStream.java

```diff
@@ -1,4 +1,45 @@
 package com.mateanna.hw5;
 
+import jdk.dynalink.Operation;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+
 public class TestStream {
+
+    private static Map<Character, BinaryOperator<Double>> operator = operations();
+
+    private static Map<Character, BinaryOperator<Double>> operations() {
+        BinaryOperator<Double> sum = (x, y) -> x + y;
+        BinaryOperator<Double> multiply = (x, y) -> x * y;
+        BinaryOperator<Double> divide = (x, y) -> x / y;
+        BinaryOperator<Double> substract = (x, y) -> x - y;
+        BinaryOperator<Double> root = (x, y) -> (x / (1 / y));
+        BinaryOperator<Double> power = (x, y) -> (Math.pow(x, (1 / y)));
+        BinaryOperator<Double> power117 = (x, y) -> ((Math.pow(((x + y) / x + 117), y)));
+
+        Map<Character, BinaryOperator<Double>> operatorMap = new HashMap<>();
+        operatorMap.put('+', sum);
+        operatorMap.put('*', multiply);
+        operatorMap.put('/', divide);
+        operatorMap.put('-', substract);
+        operatorMap.put('#', root);
+        operatorMap.put('^', power);
+        operatorMap.put('p', power117);
+
+
+        return operatorMap;
+    }
+
+    public static double calculate(double x, char operator, double y) throws NullPointerException {
```

Comment:
NullPointerException is unckecked exception. it is unnecessary to throw it.
better provide in method that there could not to be present passed operator